pkgs/tools/networking/curl/default.nix

```diff
@@ -177,16 +182,23 @@ stdenv.mkDerivation rec {
     ln $out/lib/libcurl${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libcurl-gnutls${stdenv.hostPlatform.extensions.sharedLibrary}.4.4.0
   '';
 
-  passthru = {
+  passthru = let
+    useThisCurl = attr: attr.override { curl = finalAttrs.finalPackage; };
+  in {
     inherit opensslSupport openssl;
     tests = {
-      inherit curlpp coeurl;
-      haskell-curl = haskellPackages.curl;
-      ocaml-curly = ocamlPackages.curly;
-      php-curl = phpExtensions.curl;
-      pycurl = python3.pkgs.pycurl;
+      withCheck = finalAttrs.finalPackage.overrideAttrs (_: { doCheck = true; });
```

Comment:
This one should be a channel blocker if it fails. It should be added to release.nix for the nixpkgs and nixos jobs.

Reply:
not sure i did it correctly

Reply:
```
error: attribute 'tests' missing

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-4/pkgs/top-level/release.nix:100:15:

           99|               jobs.stdenv.x86_64-linux
          100|               jobs.curl.tests.withCheck.x86_64-linux
             |               ^
          101|               jobs.cargo.x86_64-linux
```

are passthru attributes stripped?

Reply:
yeah release.nix is kind of a mess (or maybe I'm unfamiliar) and it strips all packages down using `hydraJobs`. It's supposed to help that evaluator release memory.
If you add the passthru tests as "normal" packages somewhere in `pkgs`, you won't have to deal with the release.nix code as much.

Maybe do something like this in `pkgs/test/default.nix`?

```nix

corePackageTests = recurseIntoAttrs {
  curl = recurseIntoAttrs pkgs.curl.tests;
};

```

That should give you `pkgs.tests.corePackageTests`, which you can use in the release files.